edb/pgsql/compiler/relgen.py

```diff
@@ -2823,9 +2823,12 @@ def process_set_as_enumerate(
                 or arg_expr.limit
                 or arg_expr.offset
             )
+        ) and not any(
+            f_arg.param_typemod == qltypes.TypeModifier.SetOfType
+            for _, f_arg in arg_subj.args.items()
         )
     ):
-        # Enumeration of a SET-returning function
+        # Enumeration of a non-aggregate SET-returning function
```

Comment:
OK, I think the problem is that this comment is wrong. SET-returning has nothing to do with it.